addons/mrp/models/stock_rule.py

```diff
@@ -121,6 +125,7 @@ def _prepare_mo_vals(self, product_id, product_qty, product_uom, location_id, na
             'bom_id': bom.id,
             'date_deadline': date_deadline,
             'date_planned_start': date_planned,
+            'date_planned_finished': date_planned_finished,
```

Comment:
```suggestion
            'date_planned_finished': fields.Datetime.from_string(values['date_planned']),
```
and remove the other code

Reply:
Totally agree. Done!